### ISSUE TITLE
Repeated word out of place ruins text 

### ISSUE THREAD

**Issue by brucenielson** [pymupdf/PyMuPDF#4042]
Karl Popper A World of Propensities

I'm using version 1.24.13. 

Attached is a pdf I'm trying to load as markdown. The first page reads as follows:

I shall begin with some personal memories...

It
was 54 years ago, in Prague in August 1934, that I first
attended an International Congress of Philosophy. I found it
uninspiring. But the Congress was preceded by another meeting
in Prague, organized by Otto Neurath, who had kindly invited
me to attend a 'Preliminary Conference' ('Vorkonferenz' as he
called it) which he organized on behalf of the Vienna Circle.

I came to Prague with the corrected page proofs of my book,
It
Logik der Forschung. was published three months later...essentially an Aristotelian theory at which, it appears,
Tarski and Godel arrived, independently at almost the same
It
time. was first published by Tarski in 1930, whereupon
It
Godel, of course, accepted Tarski's priority. 

Note the weird newline after the first "It" and then that word "It" becomes an unwanted artifact that breaks up the text after that several times. This problem repeats on the next page with some other word. 

I tried loading the same PDF using PyPDF and the problem goes away. (But obviously PyPDF isn't trying to convert to markdown.)

See: https://bugs.ghostscript.com/show_bug.cgi?id=708129

I can get you the file I used if desired.

**Comment by brucenielson** [comment]
Right, it must be a bug in pymupdf4llm's to_markdown call. I tried it to 'write_text'. 

I take it pymupdf4llm is not part of this repo?

**Comment by JorjMcKie** [comment]
I tried it again and still cannot reproduce the problem. Here is my result [original.zip](https://github.com/user-attachments/files/17712912/original.zip).
Environment Windows 3.11, Python 3.9, pymupdf 1.24.13, pymupdf4llm 0.0.17.

**Comment by brucenielson** [comment]
[test.txt](https://github.com/user-attachments/files/17721519/test.txt)
Here is my result with the same code.

**Comment by brucenielson** [comment]
My environment is the same as yours except I'm on a newer version of Python. i.e. 3.11 of Python.

**Comment by brucenielson** [comment]
[test.txt](https://github.com/user-attachments/files/17725177/test.txt)
Here is a new attempt. Still using Python 3.11 but now I've removed everything from the virtual environment other than:

 pip install pymupdf4llm  
 
 And I still get the weird effect. Let me try Python 3.9. It is no longer considered up to date, so there is no installer available any more. But they do have a zip and I'll try it out.

**Comment by brucenielson** [comment]
Well, I tried downloading Python 3.9 and I get the same problem. So I guess that wasn't it either.

**Comment by JorjMcKie** [comment]
I tried multiple Python versions and still cannot reproduce your problem.
Out of ideas 🤷‍♂️.

**Comment by brucenielson** [comment]
Okay, I got the problem to repeat in Colab!

https://colab.research.google.com/drive/1d3BCUI5PyV928PcJwmnx_RkvWGHGJGC9?usp=sharing

**Comment by brucenielson** [comment]
From the Colab link I just sent:

Ladies and Gentlemen,

I shall begin with some personal memories and a personal
confession of faith, and only then turn to the topic of my
lecture.

It
was 54 years ago, in Prague in August 1934, that I first
attended an International Congress of Philosophy. I found it
uninspiring. But the Congress was preceded by another meeting
in Prague, organized by Otto Neurath, who had kindly invited
me to attend a 'Preliminary Conference' ('Vorkonferenz' as he
called it) which he organized on behalf of the Vienna Circle.

I came to Prague with the corrected page proofs of my book,
It
_Logik der Forschung._ was published three months later in
Vienna, and in English 25 years later as The Logic of Scientific
_Discovery._ In Prague it was read by two Polish philosophers,
Alfred Tarski and Janina Hosiasson-Lindenbaum, the wife of
Tarski's friend and collaborator, Adolf Lindenbaum. Janina
Hosiasson and her husband were murdered when, 5 years later,
the Nazis invaded Poland and systematically exterminated what
they described as its 'Fuhrerschicht': its 'intellectual elite'. Tarski
went from Prague to Vienna where he stayed for a year and
where we became friends. Philosophically, it was the most
important friendship of my life. For I learnt from Tarski the
logical defensibility and the power of absolute and objective
truth: essentially an Aristotelian theory at which, it appears,
Tarski and Godel arrived, independently at almost the same
It
time. was first published by Tarski in 1930, whereupon
It
Godel, of course, accepted Tarski's priority. is a theory of
objective truth - truth as the correspondence of a statement
with the facts - and of absolute truth: if an unambiguously
formulated statement is true in one language, then any correct
translation of it into any other language is also true. This
theory is the great bulwark against relativism and
